config/utils_methods.py

```diff
@@ -433,3 +433,200 @@ def validate_uuid(uuid_to_test, version=4):
+        return phone_number_str  
+    else:
+        return "Mobile number has incorrect length"  
+
+from django.core.mail import EmailMessage
+def send_pdf_via_email(to_email, pdf_relative_path):
+    """Send the generated PDF as an email attachment."""
+    
+    # Construct the full path to the PDF file
+    pdf_full_path = os.path.join(MEDIA_ROOT, pdf_relative_path)
+    
+    subject = 'Your Sales Order Receipt'
+    body = 'Please find attached your sales order receipt.'
+    email = EmailMessage(subject, body, to=[to_email])
+
+    # Ensure the PDF file exists before attempting to open it
+    if not os.path.exists(pdf_full_path):
+        raise FileNotFoundError(f"The file {pdf_full_path} does not exist.")
+
+    # Read the PDF file from the provided full path
+    with open(pdf_full_path, 'rb') as pdf_file:
+        email.attach('sales_order_receipt.pdf', pdf_file.read(), 'application/pdf')
+    
+    # Send the email
+    email.send()
+
+    return "PDF sent via Email successfully."
+
+
+from config.settings import MEDIA_ROOT, MEDIA_URL
+import requests
+
+def send_whatsapp_message_via_wati(to_number, file_url):
+    """ Send the PDF file as a WhatsApp message using WATI API. """
+    result = ""
+    # Construct the full path to the file using MEDIA_ROOT
+    full_file_path = os.path.join(MEDIA_ROOT, file_url.replace(MEDIA_URL, ''))
+    # Ensure the file exists
+    if not os.path.exists(full_file_path):
+        return (f"File not found: {full_file_path}")            
+    
+    url = f'https://live-mt-server.wati.io/312172/api/v1/sendSessionFile/{to_number}'
+    
+    headers = {
+    'accept': '*/*',
+    'Authorization': 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJqdGkiOiJlMzMyNWFmNC0wNDE2LTQzYTQtOTcwNi00MGYxZDViZTM0MGQiLCJ1bmlxdWVfbmFtZSI6InJ1ZGhyYWluZHVzdHJpZXMubmxyQGdtYWlsLmNvbSIsIm5hbWVpZCI6InJ1ZGhyYWluZHVzdHJpZXMubmxyQGdtYWlsLmNvbSIsImVtYWlsIjoicnVkaHJhaW5kdXN0cmllcy5ubHJAZ21haWwuY29tIiwiYXV0aF90aW1lIjoiMDgvMjYvMjAyNCAwNjowMzozNSIsImRiX25hbWUiOiJtdC1wcm9kLVRlbmFudHMiLCJ0ZW5hbnRfaWQiOiIzMTIxNzIiLCJodHRwOi8vc2NoZW1hcy5taWNyb3NvZnQuY29tL3dzLzIwMDgvMDYvaWRlbnRpdHkvY2xhaW1zL3JvbGUiOiJBRE1JTklTVFJBVE9SIiwiZXhwIjoyNTM0MDIzMDA4MDAsImlzcyI6IkNsYXJlX0FJIiwiYXVkIjoiQ2xhcmVfQUkifQ.qFA42Ze-itghM2LXR5sZ-P9BJB84iD3oXqk5olG_kX8',
+    }
+
+    # Open the file and attach it to the request
+    with open(full_file_path, 'rb') as file:
+        files = {
+            'file': (os.path.basename(full_file_path), file, 'application/pdf'),
+        }
+        response = requests.post(url, headers=headers, files=files)
+
+        # Convert the response text to a Python dictionary
+    response_data = json.loads(response.text) 
+
+    if response_data.get("result") == True:
+        result = "PDF sent via WhatsApp successfully."
+        return result
+    else:
+        result = response_data.get('info')
+        return result
+
+import os
+import shutil
+import inflect
+from config.settings import MEDIA_ROOT
```

Comment:
Move these imports top of the file, these shouldn't be in the middle

Reply:
Moved